lib/wor/batchifier.rb

```diff
@@ -1,14 +1,20 @@
-# frozen_string_literal: true
+require_relative 'batchifier/exceptions'
+require_relative 'batchifier/interface'
+require_relative 'batchifier/strategy'
+require_relative 'batchifier/add'
+require_relative 'batchifier/no_response'
+require_relative 'batchifier/maintain_unique'
+
 module Wor
   module Batchifier
-    class WrongStrategy < StandardError; end
 
     # This module exports the function execute_in_batches, that needs a collections and
     # => optionaly a batch_size and a merge strategy. It will slice the collection and
-    # => apply the given block to all chunks and merge the results. It expects the responses
-    # => to be Hash. It can ignore them if the given stragegy is no_response
+    # => apply the chozen strategy to all chunks and merge the results. It expects the responses
+    # => to be Hash. It can ignore them if the given strategy is no_response
     def execute_in_batches(collection, batch_size: 100, strategy: :add)
-      collection.each_slice(batch_size).to_a.inject({}) do |rec, chunk|
+      strategy_class = classify_strategy(strategy)
+      collection.each_slice(batch_size).to_a.inject(strategy_class.new.base_case) do |rec, chunk|
```

Comment:
i dont get why you turn it into an array, when you should be able to do it  as an enumerator

Reply:
Yeah, this was fixed in the lazy_array PR, I dont use the to_a anymore and just work with enumerators.